### ISSUE TITLE
iOS VoiceOver cursor becomes trapped in ComboBox

### ISSUE THREAD

**Issue by WestonThayer** [adobe/react-spectrum#2108]
# 🐛 Bug Report 

1. Open Safari and visit https://react-spectrum.adobe.com/react-spectrum/ComboBox.html#example-4
2. With VoiceOver on (default settings, except I have direct touch typing enabled) navigate to the "Pick a engineering major" ComboBox
3. Double tap to activate
4. Type "a" "a" to run out of suggestions, then backspace
5. Swipe right or left

## 🤔 Expected Behavior

VoiceOver cursor navigates to suggestions list.

## 😯 Current Behavior

VoiceOver cursor is in a focus trap, cannot navigate to suggestions list.

https://user-images.githubusercontent.com/709153/125690626-1df665c0-dec2-425d-93b6-687fe6945a2e.mp4

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   | N/A latest
| Browser          | Safari
| Operating System | iOS 14.6

**Comment by devongovett** [comment]
Good catch! I verified locally. Appears to have something to do with the menu hiding and then coming back. At first I can access the options, but not after the menu hiding and reappearing. Also couldn't reproduce without direct touch typing enabled.

**Comment by yihuiliao** [comment]
Closing since I can't reproduce the issue even with direct touch typing enabled. Using iPhone 13 iOS 18.3

**Comment by WestonThayer** [comment]
I could not repro either, iPhone 13, iOS 18.5.
